fingerprint/README.md

```diff
@@ -0,0 +1,193 @@
+| **github-token**                   | `github.token`   | GitHub token to use when commenting on PR ([read more](#github-tokens)) |
+| **fingerprint-version**            | `latest`         | `@expo/fingerprint` version to install                                  |
+| **fingerprint-installation-cache** | `true`           | If the `@expo/fingerprint` should be cached to speed up installation    |
+| **fingerprint-db-cache-key**       | `fingerprint-db` | A cache key to use for saving the fingerprint database                  |
+
+And the action will generate these [outputs](#available-outputs) for other actions to do something based on current project fingerprint
+
+### Available outputs
+
+In case you want to reuse this action for other purpose, this action will set the following action outputs.
+
+| output name          | description                                                                                                                                                                                   |
+| -------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| **fingerprint-diff** | The diff between the current and the previous fingerprint. It is a JSON array of fingerprint diff. If the fingerprint does not change in between, the result diff will be an empty array `[]` |
+
+## Example workflows
+
+Here's an example workflow that uses this action to comment pull requests and add `Fingerprint:compatible` and `Fingerprint:changed` labels
+
+```yaml
+name: PR Labeler
+
+on:
+  push:
+    # REQUIRED: push main(default) branch is necessary for this action to update its fingerprint database
+    branches: [main]
+  pull_request:
+    types: [opened, synchronize]
+
+jobs:
+  fingerprint:
+    runs-on: ubuntu-latest
+    # REQUIRED: limit concurrency when pushing main(default) branch to prevent conflict for this action to update its fingerprint database
+    concurrency: fingerprint-${{ github.event_name != 'pull_request' && 'main' || github.run_id }}
+    permissions:
+      # REQUIRED: Allow comments of PRs
+      pull-requests: write # Allow comments on PRs
+      # REQUIRED: Allow updating fingerprint in acton caches
+      actions: write
+
+    steps:
+      - name: 🏗 Setup repo
+        uses: actions/checkout@v3
+
+      - name: 🏗 Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: yarn
+
+      - name: 📦 Install dependencies
+        run: yarn install
+
+      - name: Check fingerprint
+        id: fingerprint
+        uses: expo/expo-github-action/preview-build
+
+      - uses: actions/github-script@v6
+        if: ${{ github.event_name == 'pull_request' && github.steps.fingerprint.outputs.fingerprint-diff == '[]' }}
+        with:
+          script: |
```

Comment:
Would be awesome if we could add a script for users to reuse since it might clutter their workflow a bit.

For now, I think it's fine.